src/GraphQLParser.Tests/Values/GraphQLIntValueTests.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Numerics;
+using GraphQLParser.AST;
+using Shouldly;
+using Xunit;
+
+namespace GraphQLParser.Tests;
+
+public class GraphQLIntValueTests
+{
+    [Fact]
+    public void NoValue()
+    {
+        var value = new GraphQLIntValue();
+        value.Value.Length.ShouldBe(0);
+    }
+
+    [Fact]
+    public void Int()
+    {
+        var value = new GraphQLIntValue(1234567);
+        value.Value.Length.ShouldBe(7);
+        value.Value.ShouldBe("1234567");
+    }
+
+    [Fact]
+    public void Byte()
+    {
+        var value = new GraphQLIntValue((byte)42);
+        value.Value.Length.ShouldBe(2);
+        value.Value.ShouldBe("42");
+    }
+
+    [Fact]
+    public void Sbyte()
+    {
+        var value = new GraphQLIntValue((sbyte)-10);
+        value.Value.Length.ShouldBe(3);
+        value.Value.ShouldBe("-10");
+    }
+
+    [Fact]
+    public void Short()
+    {
+        var value = new GraphQLIntValue((short)-300);
+        value.Value.Length.ShouldBe(4);
+        value.Value.ShouldBe("-300");
+    }
+
+    [Fact]
+    public void Ushort()
+    {
+        var value = new GraphQLIntValue((ushort)60000);
+        value.Value.Length.ShouldBe(5);
+        value.Value.ShouldBe("60000");
+    }
+
+    [Fact]
+    public void Uint()
+    {
+        var value = new GraphQLIntValue(2247483647U);
+        value.Value.Length.ShouldBe(10);
+        value.Value.ShouldBe("2247483647");
+    }
+
+    [Fact]
+    public void Long()
+    {
+        var value = new GraphQLIntValue(-60001L);
+        value.Value.Length.ShouldBe(6);
+        value.Value.ShouldBe("-60001");
+    }
+
+    [Fact]
+    public void Ulong()
+    {
+        var value = new GraphQLIntValue(9223372036854775808UL);
+        value.Value.Length.ShouldBe(19);
+        value.Value.ShouldBe("9223372036854775808");
+    }
+
+    [Fact]
+    public void BigInt()
+    {
+        var value = new GraphQLIntValue(BigInteger.Parse("7922816251426433759354395033579228162514264337593543950335"));
+        value.Value.Length.ShouldBe(58);
+        value.Value.ShouldBe("7922816251426433759354395033579228162514264337593543950335");
+    }
+
+    [Fact]
+    public void Decimal()
+    {
+        var value = new GraphQLIntValue(1234554321m);
+        value.Value.Length.ShouldBe(10);
+        value.Value.ShouldBe("1234554321");
+    }
+
+    [Fact]
+    public void DecimalBad()
+    {
+        var ex = Should.Throw<ArgumentOutOfRangeException>(() => new GraphQLIntValue(12.3m));
+        ex.Message.Replace(",", ".").ShouldStartWith("Invalid integer number 12.3");
```

Comment:
> Replace(",", ".")

ru-RU culture locally

Reply:
I suggest we remove the decimal constructor.  The decimal data type is designed for holding floating point data.  If someone wants to create an integer value from a decimal number, it is their job to store it in a valid data type first; int, long, BigInteger, etc.

Reply:
How to deal with it here?
```c#
    public virtual GraphQLValue? ToAST(object? value)
        {
            var serialized = Serialize(value);
            return serialized switch
            {
                bool b => new GraphQLBooleanValue(b),
                byte b => new GraphQLIntValue(b),
                sbyte sb => new GraphQLIntValue(sb),
                short s => new GraphQLIntValue(s),
                ushort us => new GraphQLIntValue(us),
                int i => new GraphQLIntValue(i),
                uint ui => new GraphQLIntValue(ui),
                long l => new GraphQLIntValue(l),
                ulong ul => new GraphQLIntValue(ul),
                BigInteger bi => new GraphQLIntValue(bi),
                decimal d => Math.Truncate(d) == d ? new GraphQLIntValue(d) : new GraphQLFloatValue(d), <--------
                float f => new GraphQLFloatValue(f),
                double d => new GraphQLFloatValue(d),
                string s => new GraphQLStringValue(s),
                null => new GraphQLNullValue(),
                _ => throw new NotImplementedException($"Please override the '{nameof(ToAST)}' method of the '{GetType().Name}' scalar to support this operation.")
            };
        }
```

Reply:
Simple:

`decimal d => new GraphQLFloatValue(d),`

There is nothing incorrect about this.